src/Middleware/StaticFiles/test/UnitTests/DefaultFilesMiddlewareTests.cs

```diff
@@ -213,24 +227,34 @@ public async Task PostDirectory_PassesThrough_All(string baseUrl, string baseDir
         [InlineData("/SubFolder", @".\", "/SubFolder/")]
         [InlineData("", @".\SubFolder", "/")]
         [InlineData("", @".\SubFolder\", "/")]
-        public async Task PostDirectory_PassesThrough_Windows(string baseUrl, string baseDir, string requestUrl)
+        [InlineData("/SubFolder", @".\", "/SubFolder/", false)]
+        [InlineData("", @".\SubFolder", "/", false)]
+        [InlineData("", @".\SubFolder\", "/", false)]
+        public async Task PostDirectory_PassesThrough_Windows(string baseUrl, string baseDir, string requestUrl, bool appendTrailingSlash = true)
         {
-            await PostDirectory_PassesThrough(baseUrl, baseDir, requestUrl);
+            await PostDirectory_PassesThrough(baseUrl, baseDir, requestUrl, appendTrailingSlash);
         }
 
-        private async Task PostDirectory_PassesThrough(string baseUrl, string baseDir, string requestUrl)
+        private async Task PostDirectory_PassesThrough(string baseUrl, string baseDir, string requestUrl, bool appendTrailingSlash = true)
         {
             using (var fileProvider = new PhysicalFileProvider(Path.Combine(AppContext.BaseDirectory, baseDir)))
             {
                 var server = StaticFilesTestServer.Create(app => app.UseDefaultFiles(new DefaultFilesOptions
                 {
                     RequestPath = new PathString(baseUrl),
-                    FileProvider = fileProvider
+                    FileProvider = fileProvider,
+                    AppendTrailingSlash = appendTrailingSlash
                 }));
                 var response = await server.CreateRequest(requestUrl).GetAsync();
 
                 Assert.Equal(HttpStatusCode.NotFound, response.StatusCode); // Passed through
             }
         }
+
+        [Fact]
+        public void Options_AppendTrailingSlashByDefault()
+        {
+            Assert.True(new DefaultFilesOptions().AppendTrailingSlash);
```

Comment:
And directory browser.

Reply:
I check DirectoryBrowserOptions in DirectoryBrowserMiddlewareTests. Test logic is the same